gui/elm/Window/Onboarding/Address.elm

```diff
@@ -59,22 +59,42 @@ setError model message =
 
 dropAppName : String -> String
 dropAppName address =
-    if String.endsWith ".mycozy.cloud" address then
-        case String.split "-" address of
-            [] ->
-                ""
-
-            [ wholeaddress ] ->
-                wholeaddress
+    let
+        ( instanceName, topDomain ) =
+            case String.split "." address of
+                [] ->
+                    -- This can't happen because String.split always return a
+                    -- list with at least one element but is required by Elm.
+                    ( address, "mycozy.cloud" )
+
+                [ instance ] ->
+                    -- This should never happen as we already append
+                    -- `.mycoz.cloud` to addresses without host
+                    ( address, "mycozy.cloud" )
+
+                instance :: rest ->
+                    ( instance, String.join "." rest )
+    in
+    if String.isEmpty address then
+        ""
 
-            instanceName :: _ ->
-                instanceName ++ ".mycozy.cloud"
+    else if
+        String.endsWith "mycozy.cloud" topDomain
+            || String.endsWith "mytoutatice.cloud" topDomain
+            || String.endsWith "testcloud.toutatice.fr" topDomain
```

Comment:
I wouldn't put `testcloud.toutatice.fr` in this list (we don't have `cozy.rocks`)

Reply:
I was thinking it would be nice to smooth all their experiences as best as we can but I understand your point.